Change ngrx in mangol from forRoot to forFeature

Cosmocat opened the pull request:
This PR changes the way mangol uses ngrx from forRoot to forFeature, to avoid conflicts when mangol used as module and parent app also uses ngrx.

fegyi001 commented:
Hello @Cosmocat! Thank you for your work! 
I quickly viewed your pull request (I will do a more thorough one later), and have a couple of questions.

- You changed a lot of things in the code, not only the forFeature vs. forRoot behavior, but also wrote a lot of selectors. 
- Why did you write selectors everywhere? Was it necessary for forFeature to work, or you wrote it simply because you prefer the syntax of it?
- I can see a lot of malformatted code (unused imports, commented lines etc) which I always try to avoid to keep the code clean and well-readable.
- You also re-wrote the demo pages, using totally different layers. Did you intend to use your layers on the "official" demo pages? I would suggest to create a new demo page for your 'sandbox' that could be deleted after a merge.
- I don't think that your solution works properly. If I load a demo page (e.g. demo-layertree) and after that, change to demo-measure or demo-featureinfo, the layers are missing, sometimes I can see two OSM layers in the layertree etc. Did you test it thoroughly?

Cosmocat commented:
>You changed a lot of things in the code, not only the forFeature vs. forRoot behavior, but also wrote a lot of selectors.
>Why did you write selectors everywhere? Was it necessary for forFeature to work, or you wrote it simply because you prefer the syntax of it?


Yes, please look @ngrx/platform example app. I think it's best practice now to use selectors like that

>You also re-wrote the demo pages, using totally different layers. Did you intend to use your layers on the "official" demo pages? I would suggest to create a new demo page for your 'sandbox' that could be deleted after a merge.

I didn't mean to keep it there, i'll fix it.

I think I'm doing it wrong way since I also added methods for adding/removing layers and removed browser module and browser animation module imports, to address issue #31 

Maybe it needs to be done in different PR's

Cosmocat commented on projects/mangol/src/lib/modules/map/map.component.ts:
```diff
@@ -112,25 +113,31 @@ export class MapComponent implements OnInit, AfterViewInit, OnDestroy {
+                }
+              }) :
+              existingLayers.forEach(l => {
+                if (l.getType() === 'TILE' && tileLayers.indexOf(l) === -1) {
```
@fegyi001 
I don't think this is the best way to handle removing layers. Especially part with 'TILE', but I've been using only tile layers, that's why I have it this way. You can remove this change, but adding/removing layers dynamically is desired feature.
